### PR TITLE
Pull in re-linq and use its partial expression evaluation

### PR DESCRIPTION
**Pull request by fealebenpae**

### CODE REVIEW

File: Realm.XamarinAndroid/packages.config [hunk 1]
```diff
@@ -1,4 +1,16 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="DotNetCross.Memory.Unsafe" version="0.2.2" targetFramework="monoandroid60" />
+  <package id="Remotion.Linq" version="2.1.1" targetFramework="MonoAndroid60" />
```
Comment (AndyDentFree) [inline Realm.XamarinAndroid/packages.config]: Should we be using `targetFramework="MonoAndroid44"` instead?

File: Tests/IntegrationTests.XamarinAndroid/packages.config [hunk 2]
```diff
@@ -2,4 +2,16 @@
 <packages>
   <package id="DotNetCross.Memory.Unsafe" version="0.2.2" targetFramework="monoandroid51" />
   <package id="Fody" version="1.29.4" targetFramework="MonoAndroid44" developmentDependency="true" />
+  <package id="Remotion.Linq" version="2.1.1" targetFramework="MonoAndroid51" />
```
Comment (AndyDentFree) [inline Tests/IntegrationTests.XamarinAndroid/packages.config]: This is another `MonoAndroid` level inconsistent with `Realm.XamarinAndroid/packages.config`

**Comment by kristiandupont** [comment]
Sweet!

File: Realm.XamarinAndroid/packages.config (hunk 1 above)
Comment (fealebenpae) [inline Realm.XamarinAndroid/packages.config]: I think that's due to the value of the `TargetFrameworkVersion` property. In `Realm.XamarinAndroid` it's set to Latest, whereas in `IntegrationTests.XamarinAndroid` it's set to an older version.

File: Tests/IntegrationTests.XamarinAndroid/packages.config (hunk 2 above)
Comment (fealebenpae) [inline Tests/IntegrationTests.XamarinAndroid/packages.config]: See comment above.

File: Realm.XamarinAndroid/packages.config (hunk 1 above)
Comment (fealebenpae) [inline Realm.XamarinAndroid/packages.config]: We actually get a warning about this when building, but it's sort of benign, seeing as how we don't actually use any Android APIs.

**Comment by kristiandupont** [comment]
:+1:

**Pull request merged**